tasks/connectors/appscan_cloud/appscan_cloud_task.rb

```diff
@@ -0,0 +1,242 @@
+              description: "AppScan Cloud API Key ID" },
+            { name: "appscan_cloud_key_secret",
+              type: "api_key",
+              required: true,
+              default: nil,
+              description: "AppScan Cloud API Key Secret" },
+            { name: "appscan_cloud_applications",
+              type: "string",
+              required: true,
+              default: nil,
+              description: "A list of AppScan on Cloud Applications ID's (comma separated)" },
+            { name: "appscan_cloud_severities",
+              type: "string",
+              required: false,
+              default: nil,
+              description: "A list of ['Undetermined', 'Informational', 'Low', 'Medium', 'High', 'Critical'] (comma separated)" },
+            { name: "batch_size",
+              type: "integer",
+              required: false,
+              default: 500,
+              description: "Maximum number of issues to retrieve in batches." },
+            { name: "kenna_api_key",
+              type: "api_key",
+              required: false,
+              default: nil,
+              description: "Kenna API Key" },
+            { name: "kenna_api_host",
+              type: "hostname",
+              required: false,
+              default: "api.kennasecurity.com",
+              description: "Kenna API Hostname" },
+            { name: "kenna_connector_id",
+              type: "integer",
+              required: false,
+              default: nil,
+              description: "If set, we'll try to upload to this connector" },
+            { name: "output_directory",
+              type: "filename",
+              required: false,
+              default: "output/appscan_cloud",
+              description: "If set, will write a file upon completion. Path is relative to #{$basedir}" }
+          ]
+        }
+      end
+
+      def run(opts)
+        super
+        initialize_options
+        initialize_client
+
+        @applications.each do |application_id|
+          pos = 0
+          loop do
+            issues_data = @client.issues(application_id, pos, @batch_size, @severities)
+            issues = issues_data.fetch("Items")
+            total_issues = issues_data.fetch("Count")
+            issues.each do |issue|
+              asset = extract_asset(issue)
+              finding = extract_finding(issue)
+              definition = extract_definition(issue)
+
+              create_kdi_asset_finding(asset, finding)
+              create_kdi_vuln_def(definition)
+            end
+            print_good("Processed #{[pos + @batch_size, total_issues].min} of #{total_issues} issues for application ##{application_names[application_id]}.")
+            kdi_upload(@output_directory, "appscan_cloud_application_#{application_id}_report_#{pos}.json", @kenna_connector_id, @kenna_api_host, @kenna_api_key, @skip_autoclose, @retries, @kdi_version)
+            pos += @batch_size
+            break if pos >= total_issues
```

Comment:
would this be better listed as page size instead of batch size?  @jscalvo

Reply:
The parameter name is also `batch_size`. Do you mean renaming all references including the parameter name? I used this parameter name in all the other connectors, so must change everywhere. No problem if we define a standardized name for this commonly used param I'd agree it must be changed. Should we use page_size?

Reply:
i was using batch size for "how much we collect before we send to Kenna" and page size to be "how many rows do we request from the vendor at at time" I don't remember seeing issues with previous naming.

Reply:
I think that I was using the same parameter for both in my loops. Some don't use the page param, for instance when a report is downloaded with all the issues. Bugcrowd, Burp, JFrog are using batch_size for batches and pages. For me is a bit overkilling using 2 parameters. Maybe with Bugcrowd has more sense because page size is limited to 100, a really small number, so maybe you want to batch by 200 or 500. Anyway, I feel confusing for the end-users having to explain the differences between 2 similar parameters. As I'm new to the team I'm not aware of all the problems you had before, so, if the recommendation is always to use 2 params when possible, I'll follow it.

Reply:
@dbrothersKenna @bschmeck what do you guys think about this? I have traditionally kept the names different. Page size for how we pull from the vendor and batch size for how often we send to Kenna. I have had at least one connector where I used both for different reasons.

Reply:
@jscalvo please change this to page size as there have been cases where both are needed. too many calls to either the host api or kenna api can slow things down and there have been cases when page size (how many you get from the vendor at a time) and batch size (how many to have before sending to kenna) differs.

Reply:
OK, no problem